types/address.go

```diff
@@ -256,6 +264,21 @@ func (aa AccAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// LengthPrefixAddress prefixes the address bytes with its length, this is used
+// for variable-length components in store keys. Note: All addresses should be
+// max 255 bytes, or else this function panics.
+func LengthPrefixAddress(bz []byte) []byte {
+	if len(bz) == 0 {
+		return bz
+	}
+
+	if len(bz) > maxAddrLen {
+		panic("address length should be max 255 bytes")
```

Comment:
ouch, can we return an error instead?

Reply:
OK. I added a Must* version too, to avoid `if err!=nil {panic(err)}` in the keys.go files.

Reply:
OK, my intention was to no use `panics` at all in this context.

Reply:
I know, that's what I thought 😁. It's just a huge refactor (in all module's keys.go files).

Reply:
Yeah, I have a feeling that sooner or later we will have to do it. And would prefer sooner, rather than increasing tech debt.

Reply:
I've linked this comment to https://github.com/cosmos/cosmos-sdk/issues/7409